pkg/kn/commands/revision/describe.go

```diff
@@ -46,19 +62,225 @@ func NewRevisionDescribeCommand(p *commands.KnParams) *cobra.Command {
 				return err
 			}
 
-			printer, err := revisionDescribePrintFlags.ToPrinter()
-			if err != nil {
-				return err
+			if machineReadablePrintFlags.OutputFlagSpecified() {
+				printer, err := machineReadablePrintFlags.ToPrinter()
+				if err != nil {
+					return err
+				}
+				return printer.PrintObj(revision, cmd.OutOrStdout())
 			}
-
-			err = printer.PrintObj(revision, cmd.OutOrStdout())
-			if err != nil {
-				return err
+			printDetails, err := cmd.Flags().GetBool("verbose")
+			var service *v1alpha1.Service
+			serviceName, ok := revision.Labels[servingserving.ServiceLabelKey]
+			if printDetails && ok {
+				service, err = client.GetService(serviceName)
+				if err != nil {
+					return err
+				}
 			}
-			return nil
+			// Do the human-readable printing thing.
+			return describe(cmd.OutOrStdout(), revision, service, printDetails)
 		},
 	}
-	commands.AddNamespaceFlags(revisionDescribeCmd.Flags(), false)
-	revisionDescribePrintFlags.AddFlags(revisionDescribeCmd)
-	return revisionDescribeCmd
+	flags := command.Flags()
+	commands.AddNamespaceFlags(flags, false)
+	machineReadablePrintFlags.AddFlags(command)
+	flags.BoolP("verbose", "v", false, "More output.")
+	return command
+}
+
+func describe(w io.Writer, revision *v1alpha1.Revision, service *v1alpha1.Service, printDetails bool) error {
+	dw := printers.NewPrefixWriter(w)
+	commands.WriteMetadata(dw, &revision.ObjectMeta, printDetails)
+	WriteImage(dw, revision)
+	WritePort(dw, revision)
+	WriteEnv(dw, revision, printDetails)
+	WriteScale(dw, revision)
+	WriteConcurrencyOptions(dw, revision)
+	WriteResources(dw, revision)
+	serviceName, ok := revision.Labels[servingserving.ServiceLabelKey]
+	if ok {
+		serviceSection := dw.WriteAttribute("Service", serviceName)
+		if printDetails {
+			serviceSection.WriteAttribute("Config Gen", revision.Labels[servingserving.ConfigurationGenerationLabelKey])
```

Comment:
As we are in verbose mode anyway, could we expand this to "Configuration Generation" or maybe just "Generation" ? (compromise: "Config Generation")

Reply:
sure